back/engines/commercial/report_builder/app/services/report_builder/queries/participants.rb

```diff
@@ -0,0 +1,96 @@
+module ReportBuilder
+  class Queries::Participants < ReportBuilder::Queries::Base
+    def run_query(
+      start_at: nil,
+      end_at: nil,
+      project_id: nil,
+      resolution: 'month',
+      compare_start_at: nil,
+      compare_end_at: nil,
+      **_other_props
+    )
+      validate_resolution(resolution)
+
+      start_date, end_date = TimeBoundariesParser.new(start_at, end_at).parse
+
+      participations_in_period = participations(
+        start_date, end_date, project_id: project_id
+      )
+
+      # Time series
+      participants_timeseries = participations_in_period
+        .select("
+          count(distinct participant_id) as participants,
+          date_trunc('#{resolution}', dimension_date_created_id) as date_group
+        ")
+        .group('date_group')
+        .order('date_group')
```

Comment:
Same comment here as on your previous (similar) PR. I think this means time periods with zero participants won't appear in the time series results.

I think you said the FE handles this gracefully.

I still think we should insert these zero values, for data consistency - making the data produced by this service more robust, and avoiding needing to develop ways to find and handle some gaps in other places that might use the service.

Reply:
I don't think we need to do it. We already have the logic to do exactly this in the frontend, and it's actually pretty complex to do (considering the need to handle different resolutions etc). The code on the FE is robust and has a lot of tests, and is reused for every graph that uses time series. If we want to move this logic to the backend, I don't think we should have this logic duplicated in BE and FE, so I would have to remove it everywhere in the FE and implement it everywhere in the BE. This will take me like a full 3 days of work without any noticable difference. So I don't think it's worth it, even though I agree that it would be the ideal situation.